src/hps_align/plot/residual.py

```diff
@@ -0,0 +1,174 @@
+
+
+def single_residual_plot(p: Plotter, histo_name, title=""):
+    """Plot a single 1D residual plot for the input sensor"""
+
+    histos = [f.Get('res/'+histo_name) for f in p.input_files]
+
+    canv = r.TCanvas("c1", "c1", 2200, 2000)
+
+    if (title == ""):
+        title = histo_name.split("_")[3] + "_" + histo_name.split("_")[5]
+
+    maximum = -1.
+
+    fitList = []
+    plot_properties = []
+
+    # Normalize and get maximum after normalisation
+    for histo in histos:
+        # Normalize the histogram to unity
+        if abs(histo.Integral()) > 1e-8:
+            histo.Scale(1./histo.Integral())
+        # Get the maximum
+        if (histo.GetMaximum() > maximum):
+            maximum = histo.GetMaximum()
+
+    if ("slot") in histo_name:
+        title += "_slot"
+    elif "hole" in histo_name:
+        title += "_hole"
+
+    for ihisto in range(len(histos)):
+        p.set_histo_style(histos[ihisto], ihisto)
+
+        # Fitting
+        fitList.append(alignment_utils.make_fit(
+            histos[ihisto], "singleGausIterative", [], p.colors[ihisto]))
+
+        if (ihisto == 0):
+            histos[ihisto].GetXaxis().SetTitle(title)
+            histos[ihisto].GetXaxis().SetTitleSize(0.05)
+            histos[ihisto].GetXaxis().SetTitleOffset(1.25)
+            histos[ihisto].SetMaximum(maximum*1.5)
+            histos[ihisto].GetYaxis().SetRangeUser(0.0, maximum*1.5)
+            histos[ihisto].Draw("P")
+        else:
+            histos[ihisto].Draw("P SAME")
+
+        fitList[ihisto].Draw("SAME")
+
+        # Save fit properties
+        mu = fitList[ihisto].GetParameter(1)
+        sigma = fitList[ihisto].GetParameter(2)
+
+        plot_properties.append((" #mu=%.3f" % round(mu, 3))
+                               + (" #sigma=%.3f" % round(sigma, 3)))
+
+    leg = p.do_legend(histos, p.legend_names, 3, plot_properties)
+
+    if (leg is not None):
+        leg.Draw()
+
+    text = r.TLatex()
+    text.SetNDC()
+    text.SetTextFont(42)
+    text.SetTextSize(0.04)
+    text.SetTextColor(r.kBlack)
+    text.DrawLatex(0.16, 0.89, '#bf{#it{HPS}} Work In Progress')
+
+    canv.SaveAs(p.outdir + "/" + histo_name + p.oFext)
+
+    if p.do_HTML:
+        hw = htmlWriter(p.outdir)
+        hw.add_images(p.outdir)
+        hw.close_html()
+
+
+@Plotter.user
+def one_dim(p: Plotter):
+    """Plot the 1D residuals for the sensors in the plot listing"""
+
+    for plot in p.plot_list('1D_residual_plots'):
+        single_residual_plot(p, plot)
+
+
+@Plotter.user
+def vs_u(p: Plotter):
+    """Plot the residuals for the sensors agains u"""
+    for vol in ['top', 'bottom']:
+        names = p.plot_list('ures_vs_u_plots')[vol]['name']
+        titles = p.plot_list('ures_vs_u_plots')[vol]['title']
+        for ipl, (name, title) in enumerate(zip(names, titles)):
+            p.plot_profileY(
+                name, xtitle=title,
+                indir='res/',
+                rangeX=[-10, 10] if ipl < 2 else [-0.2, 0.2],
```

Comment:
This restriction on the x-range is overly tight (I believe). I'm not sure where it originates, but if I remove this restriction and use the empty list `[]` (i.e. allow the entire histogram to be plotted), I see the full <ures> vs u plots as normal.